tests/actor/test_power_distributing.py

```diff
@@ -232,14 +229,23 @@ async def test_power_distributor_one_user(
         channel = Bidirectional[Request, Result]("user1", "power_distributor")
 
         request = Request(
-            power=1200, batteries={106, 206}, request_timeout_sec=SAFETY_TIMEOUT
+            power=1200,
+            batteries={106, 206},
+            request_timeout_sec=SAFETY_TIMEOUT,
         )
 
         with mock.patch("asyncio.sleep", new_callable=AsyncMock):
             distributor = PowerDistributingActor(
                 mock_api, component_graph, {"user1": channel.service_handle}
             )
 
+            # Mock that all requested batteries are working.
+            distributor._battery_pool = MagicMock()
```

Comment:
Something that we are not using very often but we probably should is the `spec` argument for mocks, it will validate that the mock is only used via attributes or calls that follow some object or class. For example: `distributor._battery_pool = MagicMock(spec=BatteryPoolStatus)` should create a mock object that already has a `async_init()` attribute and it is an `AsyncMock` because it knows the signature of the method, and will also validate that the arguments of the method follow the specs.

Reply:
I didn't know that. Thanks, done